src/components/themed/TransactionListTop.tsx

```diff
@@ -181,6 +206,180 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
     Airship.show(bridge => <WalletListMenuModal bridge={bridge} tokenId={tokenId} navigation={navigation} walletId={wallet.id} />).catch(err => showError(err))
   }
 
+  handleTrade = async () => {
+    const { theme, wallet, tokenId } = this.props
+    const styles = getStyles(theme)
+
+    const buySellIconProps = { size: theme.rem(1.25), color: theme.iconTappable }
+    const sceneCurrencyCode = tokenId == null ? wallet.currencyInfo.currencyCode : wallet.currencyConfig.allTokens[tokenId].currencyCode
+
+    await Airship.show(bridge => (
+      <ModalUi4
+        bridge={bridge}
+        title={sprintf(lstrings.trade_s, sceneCurrencyCode)}
+        onCancel={() => {
+          bridge.resolve()
+        }}
+      >
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.trade_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeSwap(bridge)}
+          icon={
+            <View style={styles.singleIconContainer}>
+              <Ionicons name="swap-horizontal" size={theme.rem(2.5)} color={theme.iconTappable} />
+            </View>
+          }
+        />
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.title_plugin_buy_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeBuy(bridge)}
+          icon={
+            <View style={styles.dualIconContainer}>
+              <FontAwesomeIcon name="bank" {...buySellIconProps} />
+              <AntDesignIcon name="arrowright" {...buySellIconProps} />
+            </View>
+          }
+        />
+        <SelectableRow
+          marginRem={0.5}
+          title={sprintf(lstrings.title_plugin_sell_s, sceneCurrencyCode)}
+          onPress={() => this.handleTradeSell(bridge)}
+          icon={
+            <View style={styles.dualIconContainer}>
+              <AntDesignIcon name="arrowright" {...buySellIconProps} />
+              <FontAwesomeIcon name="bank" {...buySellIconProps} />
+            </View>
+          }
+        />
+      </ModalUi4>
+    ))
+  }
+
+  handleTradeBuy = (bridge: AirshipBridge<void>) => {
+    const { navigation, wallet, tokenId } = this.props
+    const { pluginId } = wallet.currencyInfo
+
+    navigation.navigate('buyTab', { screen: 'pluginListBuy', params: { filterAsset: { pluginId, tokenId } } })
+    bridge.resolve()
+  }
+
+  handleTradeSell = (bridge: AirshipBridge<void>) => {
+    const { navigation, wallet, tokenId } = this.props
+    const { pluginId } = wallet.currencyInfo
+
+    navigation.navigate('sellTab', { screen: 'pluginListSell', params: { filterAsset: { pluginId, tokenId } } })
+    bridge.resolve()
+  }
+
+  handleTradeSwap = (bridge: AirshipBridge<void>) => {
+    const { account, navigation, wallet, tokenId } = this.props
```

Comment:
I won't require it in this PR, but this should have been a test driven developed routine. Might have actually been faster to write and test than doing it that way.